src/components/AccountSummary/AccountSummary.fees.js

```diff
@@ -25,48 +27,64 @@ const getColumns = ({
       id: 'makerFee',
       name: makerFee > 0 ? 'column.maker_fees' : 'column.maker_rebate',
       width: 100,
-      renderer: () => (
-        <Cell tooltip={getTooltipContent(formattedMakerFee, t)}>
-          {formatAmount(makerFee * 100, { color: getColor(makerFee), minDigits: 2 })}
-          %
-        </Cell>
-      ),
+      renderer: () => {
+        if (isLoading) return getCellLoader(14, 72)
+        if (isNoData) return getCellNoData(t('column.noResults'))
+        return (
+          <Cell tooltip={getTooltipContent(formattedMakerFee, t)}>
+            {formatAmount(makerFee * 100, { color: getColor(makerFee), minDigits: 2 })}
+            %
+          </Cell>
+        )
+      },
       copyText: () => formattedMakerFee,
     },
     {
       id: 'takerFeeCrypto',
       name: takerFeeToCrypto > 0 ? 'column.taker_fees_crypto' : 'column.taker_rebate_crypto',
       width: 140,
-      renderer: () => (
-        <Cell tooltip={getTooltipContent(formattedTakerFeeToCrypto, t)}>
-          {formatAmount(takerFeeToCrypto * 100, { color: getColor(takerFeeToCrypto), minDigits: 2 })}
-          %
-        </Cell>
-      ),
+      renderer: () => {
+        if (isLoading) return getCellLoader(14, 72)
+        if (isNoData) return getCellNoData()
+        return (
+          <Cell tooltip={getTooltipContent(formattedTakerFeeToCrypto, t)}>
+            {formatAmount(takerFeeToCrypto * 100, { color: getColor(takerFeeToCrypto), minDigits: 2 })}
+            %
+          </Cell>
+        )
+      },
       copyText: () => formattedTakerFeeToCrypto,
     },
     {
       id: 'takerFeeFiat',
       name: takerFeeToFiat > 0 ? 'column.taker_fees_fiat' : 'column.taker_rebate_fiat',
       width: 140,
-      renderer: () => (
-        <Cell tooltip={getTooltipContent(formattedTakerFeeToFiat, t)}>
-          {formatAmount(takerFeeToFiat * 100, { color: getColor(takerFeeToFiat), minDigits: 2 })}
-          %
-        </Cell>
-      ),
+      renderer: () => {
+        if (isLoading) return getCellLoader(14, 72)
+        if (isNoData) return getCellNoData()
+        return (
+          <Cell tooltip={getTooltipContent(formattedTakerFeeToFiat, t)}>
+            {formatAmount(takerFeeToFiat * 100, { color: getColor(takerFeeToFiat), minDigits: 2 })}
+            %
+          </Cell>
+        )
+      },
       copyText: () => formattedTakerFeeToFiat,
     },
     {
       id: 'takerFeeStable',
       name: takerFeeToStable > 0 ? 'column.taker_fees_stable' : 'column.taker_rebate_stable',
       width: 140,
-      renderer: () => (
-        <Cell tooltip={getTooltipContent(formattedTakerFeeToStable, t)}>
-          {formatAmount(takerFeeToStable * 100, { color: getColor(takerFeeToStable), minDigits: 2 })}
-          %
-        </Cell>
-      ),
+      renderer: () => {
```

Comment:
proposal: we could extract the `renderer` logic and refactor it making it accept props for the different cases

Reply:
Good idea!
But we have a lot of different reports/tables configs and such global refactoring better be done step by step as a separate task. Added a [ticket](https://app.asana.com/0/1163495710802945/1206737300284991) for this optimization proposal.